.test/unit/test_versioning_advanced.py

```
@@ -0,0 +1,532 @@
+    WORKFLOW_PATH = Path("/home/jailuser/git/.github/workflows/versioning.yml")
+
+    def test_workflow_prevents_infinite_loops(self):
+        """Test workflow prevents infinite triggering loops"""
+        content = self.WORKFLOW_PATH.read_text()
+        # Should ignore VERSION files to prevent infinite loops
+        assert 'paths-ignore' in content, "Workflow should have paths-ignore"
+        assert 'VERSION' in content, "Workflow should ignore VERSION file changes"
+
+    def test_workflow_commit_message_has_skip_marker(self):
+        """Test workflow ensures commits have skip marker"""
+        content = self.WORKFLOW_PATH.read_text()
```

Ruff (0.14.14)

[error] Local variable `content` is assigned to but never used

Remove assignment to unused variable `content`

(F841)